x-pack/legacy/plugins/ml/public/jobs/jobs_list/components/job_filter_bar/job_filter_bar.js

```diff
@@ -145,6 +162,17 @@ class JobFilterBarUI extends Component {
             onChange={this.onChange}
             className="mlJobFilterBar"
           />
+          }
+          {selectedId !== undefined &&
+          <EuiSearchBar
+            box={{
+              incremental: true,
+            }}
+            defaultQuery={selectedId}
```

Comment:
Because `EuiSearchBar` only sets the default query when it's first mounted, if the value isn't available immediately it doesn't show up. Since in this component we first have to pull the value from the url then update the state, the default query won't show up if the search bar has already been rendered once.